test/currency-test.js

```diff
@@ -54,6 +56,16 @@ describe('Currency', function() {
       assert(r.is_valid());
       assert.strictEqual('1D2', r.to_json());
     });
+    it('from_json("1").to_human()', function() {
+      var r = currency.from_json('1');
+      assert(r.is_valid());
+      assert.strictEqual(1, r.to_json());
+    });
+    it('from_json("#$%").to_human()', function() {
+      var r = currency.from_json('#$%');
```

Comment:
Rippled is going to fail on a currency like this right? So this is only to support older currencies that are in the ledger somewhere. Do we want to make this behavior explicit with an options parameter? If the option is provided, currencies like this won't trigger an error, otherwise they will. That way you have to be explicit and can't create an invalid currency in ripple-lib unless you know what you're doing